src/main/java/org/sentrysoftware/ipmi/core/coding/security/IntegrityHmacMd5_128.java

```diff
@@ -0,0 +1,85 @@
+package org.sentrysoftware.ipmi.core.coding.security;
+
+/*-
+ * ╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲
+ * IPMI Java Client
+ * ჻჻჻჻჻჻
+ * Copyright 2023 Verax Systems, Sentry Software
+ * ჻჻჻჻჻჻
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * ╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱╲╱
+ */
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * HMAC-MD5-128 integrity algorithm.
+ */
+public class IntegrityHmacMd5_128 extends IntegrityAlgorithm {
+
+    public static final String ALGORITHM_NAME = "HmacMD5";
+    private Mac mac;
+
+    /**
+     * Initiates HMAC-MD5-128 integrity algorithm.
+     *
+     * @throws NoSuchAlgorithmException
+     *             - when initiation of the algorithm fails
+     */
+    public IntegrityHmacMd5_128() throws NoSuchAlgorithmException {
+        mac = Mac.getInstance(ALGORITHM_NAME);
+
+    }
+
+    @Override
+    public void initialize(byte[] sik) throws InvalidKeyException {
```

Comment:
Try to factorize the code here too for the three integrity algorithm implementations using a getAlgorithmName() method.